### DISCUSSION TITLE
Contributor call: 5th of February 2022

### DISCUSSION THREAD

**Discussion by darigovresearch** [freesewing/freesewing#1699] [Support] [answered]
You are all invited to the FreeSewing Contributor Call.
Please leave a comment here or [on Discord](https://discord.freesewing.org/) to add things to the agenda.
This issue will be updated after the call with notes and to-dos.

### About
FreeSewing's contributor call is an informal hangout/chat/meeting that is open to all FreeSewing contributors and users alike.
The call is organized/hosted by @tangerineshark and @lucibytes on our Discord server.
The call does not have a fixed structure or agenda items. Everyone is free to suggest agenda items, ask questions, or bring ideas.

Details:

Location: https://discord.freesewing.org (meeting room: 🔊voice chat)
Date: Saturday February the 5th, 2022 16:00 UTC, which is:
Los Angeles: 09:00
Chicago: 11:00
New York: 12:00
São Paulo: 14:00
London: 17:00
Paris: 18:00
Johannesburg: 19:00
Moscow: 20:00
Mumbai: 22:30
Hong Kong: 01:00 on Sunday
Tokyo: 02:00 on Sunday
Sydney: 04:00 on Sunday
Or check the bot in Discord! #meetings

### Agenda
- This thing about bubbles: https://github.com/freesewing/freesewing/discussions/1781
- Related: options for pattern release ✨ fanfare ✨ 
- Can @joostdecock explain the thing where everything is monorepo??

**Comment by joostdecock** [comment]
Don't know about notes for the call, but I would suggest writing them as a comment, rather than updating this initial description.

That way, we can mark that as the answer (and have answered being an indicator that we've moved on to other things)

**Reply by l3xander** [reply]
OK, sounds good! I def don't mind having my answer count go up :P

**Reply by darigovresearch** [reply]
Yeah would agree that comments are great & it's helpful to use the mark as answer, will be happy to mark it as the answer whenever you submit the comment!

**Reply by l3xander** [reply]
Thanks! I found out I can mark my own messages as answers

**Comment by l3xander** [comment, marked as answer]
### Notes
- @nataliasayang is working on Aaron with waist shaping! ([Picture from the Discord.](https://cdn.discordapp.com/attachments/757631205804998759/939569752395898970/unknown.png))
- @joostdecock explained the new “super monorepo”:
    - The issue with multiple repositories is, when we need to update something, e.g. a pattern, there are a lot of steps involved if the code of the website is in a different repository. Having everything in one place makes it easier to update things.
    - The markdown and .dev site code are already moved to the monorepo.
    - The downside is that avoiding conflicts can be hard; changing something may be easier, but it’s also easier to break something else in the process.
- @joostdecock also explained the plans for the new .dev site:
    - The documentation on the .dev site is geared towards “do your own thing”, but a lot of people want to be part of FreeSewing, so the idea is to update the documentation to make it easier for people to do so.
    - Currently, there are a lot of steps involved when working on a pattern. Say you’re busy with Simon, but you realize there’s a problem with the pattern it depends on, Brian. Then to fix Brian, you’d have to build the complete environment for the Brian pattern, which takes extra time and effort.
    - The new development environment will be built from the actual source code, instead of the published already compiled code. There will be *one* development environment for *all* patterns, instead of having to download individual development environments.
- @bobgeorgethe3rd brought up the idea of pattern ambassadors.
    - Context: we recently started [the “all contributors” approach](https://freesewing.org/blog/all-contributors/), which exists to recognize all contributors, not just those who contribute with code. The list of contributors is shown on the Github page and .dev site.
    - @bobgeorgethe3rd has [the idea to use this for patterns](https://github.com/freesewing/freesewing/discussions/1781) too, that the icons (”bubbles”) of the developers, designers, contributors and ambassadors are shown on the page of the pattern they worked on.
    - A pattern ambassador is someone who makes sure the pattern is still working after new versions and that it has complete documentation. Basically making sure the pattern doesn’t get neglected. This ‘position’ does not need any coding experience, which makes it an accessible way for new people to contribute.
    - @joostdecock wants to add this to the new .org site, as well as the idea to make it easier for people to ‘take a look under the hood’, now a lot of stuff is hidden under, for example, advanced settings.  
    Also, who is going to manage this ambassador program?
    - @tangerineshark mentioned the issue of, how are we going to recruit people? It needs to be obvious that this ‘position’ exists. A Discord announcement is one of the options.
    - @nataliasayang mentioned that pattern ambassadors need to know who to contact when they spot problems they don’t know how to fix.
    - @joostdecock followed up on that with, if there’s a commonly made mistake in a pattern, there can be a warning on the site saying “if X goes wrong, do Y”. This can also be part of a pattern ambassador’s ‘job’.
- @tangerineshark followed up on this with the mention that some patterns could use a bit more ‘fanfare’ - a bit more love.
    - With open-source projects, things get released and are then added to and completed over time. With for-profit projects, everything is released at the same time. We don’t/can’t do this, and sometimes patterns get forgotten as a result, maybe we can try to find a way to encourage folks to pick up roles like making showcases or documentation.
    - @joostdecock doesn’t mind having a “hidden gem” pattern, and often content comes sooner or later, like [James who did videography on making a Teagan shirt](https://www.youtube.com/watch?v=UXQAhaf7-HE). Things grow over time and ‘marketing’ isn't FreeSewing’s strong suit.
    - @bobgeorgethe3rd mentioned that we could collab with marketing students who’d need it for their CV. But they could be hard to find, and in @joostdecock’s words, we may not be cool enough for them.
    - We can still do a non-commercial, ‘mini’ version of this, though, where we ask people to make showcases and documentation.
- @joostdecock on a new development update, of which @nicholasdower did a large part: there will be the option to conditionally display pattern options.
    - Which means: if, for example, you make Paco pants and click that you don’t want back pockets, the pattern options for configuring back pockets will not be shown.
    - Code and documentation is done, has to be implemented still.
- Another thing in development is the updated percentage option: it will show both the percentage and centimeters/inches of an option, and it will be possible to enter the desired number of centimeters/inches, which will automatically be converted to the right percentage. The platform supports it, but it has to be configured per pattern, so that will take a while.
- @nicholasdower asked if there will be a demo on the new .dev site, as he couldn’t get the folder freesewing.lab to run. @joostdecock replied that it’s too early to release a demo, but it will be released somewhere in the following months. (Possibly before the next meeting.)
- @bobgeorgethe3rd asked about info on the upcoming newsletter.
    - @tangerineshark has an interview from @eleonoremilbou we can add.
    - We can let people know about the pattern ambassadors plan.
    - Theoretically, the deadline is mid-march. But @joostdecock added that @ttimearl doesnt proofread anymore, so it could be later. It is set for April 1st, which also means, joke articles!

### To-do's
- Throw an announcement out on Discord about pattern ambassadors
- Encourage people to make showcases (Discord)
- Pattern ambassador plan on roadmap? (@joostdecock)

### 'Bonus'
Compact list of things that would be part of a pattern ambassador's 'job'
- Check if the documentation makes sense and keep it up-to-date.
- Check if drafting it works.
- Check if a new version doesn't break the pattern, if it does: contact a developer if they're not a developer themself and get a fix or warning on the page.
Which patterns need a pattern ambassador? Well, more or less, all of them.

**Comment by joostdecock** [comment]
Closing to increase visibility of active discussions
